### ISSUE TITLE
Bug: `loginExpirationInDays`

### ISSUE THREAD

**Issue by mariorodriguespt** [meteor/meteor#3656]
I want to allow users to check "Remember me" at the login page. According to the [documentation](http://docs.meteor.com/#/full/accounts_config), I can run this code anywhere but I don't see this behaviour working in the browser.

Running on the client:

```
Accounts.config({
            loginExpirationInDays: null
        });
```

When I check localStorage I get: Meteor.loginTokenExpires : Sat Feb 07 2015 06:34:22 GMT+0000 (WET)

It doesn't matter which number I set on the client, the server defines the expiration date. Bug in the docs?
When I don't set a configuration on the server, I get the message on the console: `Accounts.config was called on the client but not on the server; some configuration options may not take effect.`

When I try to set a new value for `loginExpirationInDays` I get the error: `Error: Can't set 'loginExpirationInDays' more than once`

I want to implement the "remember me" feature but It seems like Meteor doesn't provides it. 

If I was able to set the expiration date on the client or pass some flag to the login method, I could implement it. I also noticed that `services.resume.loginTokens` contains a creation date `when` but I don't see an expiration date. So, does a loginToken expires based on `when` + `loginExpirationInDays` date?

How can I safely implement the `Remember me` feature?
Am I missing something?

**Comment by glasser** [comment]
You're supposed to run this code in both the client and the server.  Question: when you run it only on the client, if you look in the error console in your app, do you see a message like:

"Accounts.config was called on the client but not on the server; some configuration options may not take effect."

?

(I (a) am curious if the thing we did to mitigate this issue stopped even sort of working and (b) agree that this at least needs to be better documented.)

**Comment by glasser** [comment]
and (c) maybe should reconsider the earlier decision that made calling `Accounts.config` on the client not just be an error.  (And change the code so that it only needs to be run on the server.)

**Comment by mariorodriguespt** [comment]
Regarding: ""Accounts.config was called on the client but not on the server; some configuration options may not take effect." - Yes, I got this message also when I was trying to make it worked with separated configs..

Wouldn't it make sense to decide the login token expiration based on the client input and fallback to the server config if the expiration isn't provided? I would be happy to have this because of the "remember me" feature :)

Are you guys accepting pull requests for the accounts package for this particular issue/lack of feature?

**Comment by glasser** [comment]
This is a security feature, so it really should be decided on the server.  Unfortunately, the UX tradeoffs we made when designing `Accounts.config`, along with poor documentation, makes it not obvious that you should run it in both places.  To make matters worse, some keys (oauthSecretKey) can only be specified on the server.  We should probably change `Accounts.config` to only work on the server and to use `_meteor_runtime_config_` to publish relevant configuration to the client.  This unfortunately would be a backwards-incompatible change.

Definitely excited for a docs PR that improves the docs.  If you understand how to use `_meteor_runtime_config_` then I'd look at that too.

Does the feature work for you if you put the `Accounts.config` call in a file that runs on both server and client for now, though?

**Comment by mariorodriguespt** [comment]
Yes, it works but, as far as I know, it will setup a common expiration date for all the login tokens which isn't the behaviour that I'm looking for.

**Comment by glasser** [comment]
Can you please follow our guidelines on how to report a bug: https://github.com/meteor/meteor/wiki/Contributing-to-Meteor#reporting-a-bug-in-meteor

It will be a lot easier for us to not be talking past each other if I can literally run the code that you're concerned about. It seems like there are several issues here and it's hard to parse them out without actually having a reproduction.

**Comment by mariorodriguespt** [comment]
Sure, I'll created a repo to demonstrate this and I'll paste the URL here.

**Comment by pors** [comment]
Might be related to https://github.com/meteor/meteor/issues/5121

**Comment by CTimmerman** [comment]
Client:

Meteor.loginWithPassword(username, password, function(error, result){
	if(!error){
		// BUL-514. Session logins are broken: https://github.com/meteor/meteor/issues/3656
		Meteor.call("user_remember_login", remember, function(error){
			try{
				if(remember) Accounts.config({loginExpirationInDays: 3650})
				else Accounts.config({loginExpirationInDays: null})
			}catch(ex){
				console.log(ex)
			}
		})
	}
}


Server:

Meteor.methods({
	user_remember_login: function(remember){  // BUL-514
		try{
			// Probably broken: https://github.com/meteor/meteor/issues/3656
			if(remember) Accounts.config({loginExpirationInDays: 3650})
			else Accounts.config({loginExpirationInDays: null})
		}catch(ex){
			console.log(ex)
		}
	}
)


Action:

Login.


Result:

Both client and server say "Error: Can't set `loginExpirationInDays` more than once", and token timout is 90 days no matter remember being true or false.


Expected result:

Joe Sixpack can log in at a public terminal, close the browser or terminal session, and not worry about people stealing his identity because he didn't check the "Remember session" box when he logged in.

**Comment by osv** [comment]
@CTimmerman 
`Accounts.config({loginExpirationInDays ` set system-wide expiration, not per user as I remember.

**Comment by sasikanth513** [comment]
Is there anyway to set the expiry per user per session?

**Comment by hwillson** [comment]
I've opened https://github.com/meteor/docs/issues/158 to track the documentation request in https://github.com/meteor/meteor/issues/3656#issuecomment-73805579. That should take care of the original issue.
